src/libsystemd/sd-journal/journal-file.c

```diff
@@ -3931,16 +3929,16 @@ int journal_file_copy_entry(JournalFile *from, JournalFile *to, Object *o, uint6
                 else
                         xor_hash ^= le64toh(u->data.hash);
 
-                items[i].object_offset = htole64(h);
-                items[i].hash = u->data.hash;
+                items[n_items].object_offset = htole64(h);
+                items[n_items].hash = u->data.hash;
+                n_items++;
```

Comment:
maybe:

```c
items[n_items++] = (EntryItem) {
        .object_offset = …,
        .hash = …,
};
```

we love structured initialization in the systemd tree

Reply:
Added a separate commit that introduces structured initialization everywhere in journal-file.c